src/test/java/org/opensearch/knn/index/engine/InternalKNNEngineTests.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.engine;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
+import org.apache.lucene.index.SegmentReader;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.common.lucene.index.OpenSearchLeafReader;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+import org.opensearch.index.engine.Engine;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.ThreadLeakFiltersForTests;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.plugin.JVectorKNNPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.Netty4ModulePlugin;
```

Comment:
Path change is breaking the build and the automatic backport. Will create a manual one.